pyvo/dal/vosi.py

```diff
@@ -32,17 +32,43 @@ def _get_endpoint_candidates(self, endpoint):
         return [f'{curated_baseurl}/{endpoint}', url_sibling(curated_baseurl, endpoint)]
 
     def _get_endpoint(self, endpoint):
-        for ep_url in self._get_endpoint_candidates(endpoint):
+        """Attempt to connect to service endpoint"""
+        attempted_urls = []
+
+        try:
+            candidates = self._get_endpoint_candidates(endpoint)
+        except Exception as e:
```

Comment:
Can this be more specific, or at the end of the day all exceptions are expected to be invalid URLs?

Reply:
I've added the exceptions I'd expect to be thrown here (`ValueError`, `TypeError` & `AttributeError`). Let me know if you think I've missed any